packages/datadog-plugin-jest/src/index.js

```diff
@@ -0,0 +1,197 @@
+      [CI_WORKSPACE_PATH]: GITHUB_WORKSPACE,
+      [GIT_BRANCH]: GITHUB_REF,
+      [GIT_COMMIT_SHA]: GITHUB_SHA
+    }
+  }
+  return {}
+}
+
+async function getGitInformation () {
+  const sanitizeStdout = (stdout) => stdout.replace(/(\r\n|\n|\r)/gm, '')
+  try {
+    const [{ stdout: readRepository }, { stdout: readBranch }, { stdout: readCommit }] = await Promise.all([
+      promiseExec('git ls-remote --get-url'),
+      promiseExec('git branch --show-current'),
+      promiseExec('git rev-parse HEAD')
+    ])
+    return {
+      repository: sanitizeStdout(readRepository),
+      branch: sanitizeStdout(readBranch),
+      commit: sanitizeStdout(readCommit)
+    }
+  } catch (e) {
+    return {
+      repository: '',
+      branch: '',
+      commit: ''
+    }
+  }
+}
+
+function finishStartedSpans () {
+  global.tracer
+    .scope()
+    .active()
+    ._spanContext._trace.started.forEach((span) => {
+      span.finish()
+    })
+}
+
+function setTestStatus (status) {
+  global.tracer.scope().active().setTag(TEST_STATUS, status)
+}
+
+function addTestMetadata ({ testName, testSuite }) {
+  global.tracer
+    .scope()
+    .active()
+    .addTags({
+      [TEST_TYPE]: 'test',
+      [TEST_NAME]: testName,
+      [TEST_SUITE]: testSuite
+    })
+}
+
+module.exports = function (BaseEnvironment) {
+  return class DatadogJestEnvironment extends BaseEnvironment {
+    constructor (config, context) {
+      super(config, context)
+      this.testSuite = context.testPath.replace(`${config.rootDir}/`, '')
+      this.rootDir = config.rootDir
+    }
+    async setup () {
+      if (!global.tracer) {
+        const ciMetadata = getCIMetadata()
+        const { repository, branch, commit } = await getGitInformation()
+        global.tracer = require('../../dd-trace').init({
+          sampleRate: 1,
+          flushInterval: 1,
+          startupLogs: false,
+          ingestion: {
+            sampleRate: 1
+          },
+          tags: {
+            ...ciMetadata,
+            [GIT_COMMIT_SHA]: commit,
+            [GIT_BRANCH]: branch,
+            [GIT_REPOSITORY_URL]: repository,
+            [BUILD_SOURCE_ROOT]: this.rootDir,
+            [TEST_FRAMEWORK]: 'jest'
+          }
+        })
+      }
+
+      return super.setup()
+    }
+    async teardown () {
+      await new Promise((resolve) => {
+        global.tracer._tracer._exporter._writer.flush(resolve)
+      })
+      return super.teardown()
+    }
+
+    async handleTestEvent (event) {
+      if (event.name === 'test_skip' || event.name === 'test_todo') {
+        global.tracer.trace(
+          'jest.test',
+          { type: 'test', resource: `${event.test.parent.name}.${event.test.name}` },
+          (span) => {
+            span.addTags({
+              [TEST_TYPE]: 'test',
+              [TEST_NAME]: event.test.name,
+              [TEST_SUITE]: this.testSuite,
+              [TEST_STATUS]: 'skip'
+            })
+          }
+        )
```

Comment:
For skipped tests, you probably don't need all the extra machinery of `trace`, and can instead do something like:

```js
global.tracer.startSpan(
  'jest.test',
  {
    tags: {
      type: 'test',
      resource: `${event.test.parent.name}.${event.test.name}`,
      [TEST_TYPE]: 'test',
      [TEST_NAME]: event.test.name,
      [TEST_SUITE]: this.testSuite,
      [TEST_STATUS]: 'skip'
    }
  }
).finish()
```

Reply:
very cool!